cmd/eksctl-anywhere/cmd/upgradeplancluster.go

```diff
@@ -89,13 +91,28 @@ func (uc *upgradeClusterOptions) upgradePlanCluster(ctx context.Context) error {
 		return err
 	}
 
-	componentChangeDiffs := eksaupgrader.EksaChangeDiff(currentSpec, newClusterSpec)
-	if componentChangeDiffs == nil {
-		componentChangeDiffs = &types.ChangeDiff{}
+	componentChangeDiffs := &types.ChangeDiff{}
+
+	if newClusterSpec.Cluster.IsSelfManaged() {
+		client, err := deps.UnAuthKubeClient.BuildClientFromKubeconfig(managementCluster.KubeconfigFile)
+		if err != nil {
+			return err
+		}
+
+		currentManagementComponentsVersionsBundle, err := cluster.GetManagementComponentsVersionsBundle(ctx, client, currentSpec.Cluster)
+		if err != nil {
+			return err
+		}
+
+		newVersionsBundle := &newClusterSpec.Bundles.Spec.VersionsBundles[0]
```

Comment:
I would prefer if we had a method where you pass a Bundles and it gives you the management component versions bundle
`cluster.ManagementComponentsVersionsBundle(bundles *releasev1.Bundles) *releasev1.VersionsBundle`

That way there is only one place making this assumption
And you can easily change it later.